lib/util/identifier.js

```diff
@@ -17,32 +21,50 @@ const path = require("path");
  * @returns {string} relative path in request style
  */
 const absoluteToRequest = (context, maybeAbsolutePath) => {
-	if (/^\/.*\/$/.test(maybeAbsolutePath)) {
-		// this 'path' is actually a regexp generated by dynamic requires.
-		// Don't treat it as an absolute path.
-		return maybeAbsolutePath;
+	if (maybeAbsolutePath[0] === "/") {
+		if (
+			maybeAbsolutePath.length > 1 &&
+			maybeAbsolutePath[maybeAbsolutePath.length - 1] === "/"
+		) {
+			// this 'path' is actually a regexp generated by dynamic requires.
+			// Don't treat it as an absolute path.
+			return maybeAbsolutePath;
+		}
+
+		const querySplitPos = maybeAbsolutePath.indexOf("?");
+		let resource =
+			querySplitPos === -1
+				? maybeAbsolutePath
+				: maybeAbsolutePath.slice(0, querySplitPos);
+		resource = path.posix.relative(context, resource);
+		if (!resource.startsWith("../")) {
+			resource = "./" + resource;
+		}
+		return querySplitPos === -1
+			? resource
+			: resource + maybeAbsolutePath.slice(querySplitPos);
 	}
-	if (/^[a-zA-Z]:\\/.test(maybeAbsolutePath)) {
-		const splitPath = maybeAbsolutePath.split("?", 2);
```

Comment:
This code was actually buggy. Split with limit throws away all parts after the second match. This was not intended...